components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/bapi/Axis2RFCHandler.java

```diff
@@ -70,47 +70,55 @@ public void handleRequest(JCoServerContext jCoServerContext,
         if (log.isDebugEnabled()) {
             log.debug("New BAPI function call received");
         }
-        workerPool.execute(new BAPIWorker(jCoServerContext, jCoFunction));
+        //TODO experimental code - validate
+        String xml = jCoFunction.toXML();
+        ByteArrayInputStream bais = new ByteArrayInputStream(xml.getBytes());
```

Comment:
Shall we perform this in the BAPIWorker thread?

Reply:
made the changes